aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -285,18 +371,29 @@ Tensor& copy_sparse_csr_(Tensor& self, const Tensor& src, bool non_blocking) {
   TORCH_CHECK(
       self.sizes() == src.sizes(),
       "copy_sparse_csr_: only same size tensors are supported.");
-  TORCH_CHECK(
-      self.is_sparse_csr() && src.is_sparse_csr(),
-      "copy_sparse_csr_: copy between different layouts is not supported. Found self type = ",
-      self.toString(),
-      " and src type = ",
-      src.toString());
+  switch (src.layout()) {
```

Comment:
I'd write a helper function for this pattern. It seems to come up a lot.

Reply:
Yes, this is a common pattern. However, the content of `case` blocks varies in different places and I am not sure what should be the corresponding helper function. A function that takes four lambda functions as arguments? So, can you be more specific on the helper function idea?

Reply:
Resolving as a no-op: the switch statement is the optimal approach.

Reply:
We have https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Dispatch.h to write generic code for each case, we could add something similar for layouts.